### PR TITLE
Fixes #2095

### LINKED ISSUE #2095: $elem.chosen('destroy') initializes chosen
If you perform a chosen('destroy') on an element for which chosen has not yet been initialized, the code does the opposite and initializes chosen on that element. I would expect that in such a case rather nothing would happen.

	$.fn.extend({
			 chosen: function(options) {
		  if (!AbstractChosen.browser_is_supported()) {
			return this;
		  }
		  return this.each(function(input_field) {
			var $this, chosen;
			$this = $(this);
			chosen = $this.data('chosen');
			if (options === 'destroy' && chosen) { // *)
			  chosen.destroy();
			} else if (!chosen) {
			  $this.data('chosen', new Chosen(this, options));
			}
		  });
		}
	});

*) For elements not initialized with chosen, the variable `chosen` will be `false`, hence the whole condition is false, bringing the execution to the next if. And because said variable `chosen` is false that if statement's block is executed, performing `new Chosen(this, options)`, and by that initializing chosen.

### PR DESCRIPTION
**Pull request by yallups**
If option passed is 'destroy' then only call the #destroy method and terminate.
This fixes the issue where a new chosen instance is created unintentionally.

### CODE REVIEW

**Comment by tjschuck** [comment]
@yallups Can you squash this down to one commit?  [Instructions](http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request).

**Comment by tjschuck** [comment]
This seems okay to me as far as patching the current implementation goes.

@koenpunt You self-assigned #2095 a while ago -- you want to :+1: / :-1: this?

**Comment by yallups** [comment]
Welp. I messed that up somehow. let me try again.

**Comment by yallups** [comment]
Squashed! Thanks @tjschuck for the help.

**Comment by koenpunt** [comment]
It should function like this, so I'm fine with the merge. But I'm still not happy with how the `if/else/return` looks. Also the duplicate conditions. But I'm also not sure what would be the best logic.
 But that could be a later change.

Thanks @yallups!

**Comment by koenpunt** [comment]
And a small note on the commit message. The textual "subject" is often of more value than the issue id.

So where you now have:

> Fixes #2095
If option passed is 'destroy' then only call the #destroy method and terminate.

You're better of the other way around:

> If option passed is 'destroy' then only call the #destroy method and terminate.
Fixes #2095

**Comment by yallups** [comment]
Cool. Great note. thanks for that

**Comment by tjschuck** [comment]
Thanks @yallups and @koenpunt :heart: :heart: :heart_eyes_cat: :heart: !

Draft release notes for the next version have been updated accordingly.

**Comment by koenpunt** [comment]
For example, bootstrap uses the following plugin initializer:

```js

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.alert')

      if (!data) $this.data('bs.alert', (data = new Alert(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }
```

They initialize if not yet initialized, and after that they call the method. 

So for Chosen that would mean that calling `chosen('destroy')` on a non Chosen instance would create one and immediately destroys it.

Not sure if that's the way to go either..
